### PR TITLE
Streamline the OpenStack provider README

### PR DESCRIPTION
**Pull request by tomassedovic**
#### What does this PR do?

This moves all the extra configuration options and deployment notes to a
new `advanced-configuration.md` file and keeps the README much shorter.

The README now presents the simplest workflow with minimal configuration
and manual steps on part of the deployer.

The advanced configuration is in need of a little more cleanup, but we
can do that in another pull request.

#### How should this be manually tested?
Run an end-to-end deployment using the steps in the README.

Then validate the cluster by following the steps in the Advanced Configuration page.

#### Is there a relevant Issue open for this?
n/a

#### Who would you like to review this?
cc: @tzumainn @Tlacenka  @bogdando PTAL

### CODE REVIEW

**Comment by tomassedovic** [comment]
@danielmellado and @mbruzek I'd appreciate if you could take a look at this reworked readme.

You're relatively fresh to the project -- would this be an improvement for you if this was your first contact with our docs?

File: playbooks/provisioning/openstack/README.md [hunk 1]
```diff
@@ -1,608 +1,260 @@
...
-(feel free to replace `$PWD` with an actual path to your inventory and
-checkouts, but note that relative paths don't work)
+```bash
+$ vi inventory/group_vars/all.yml
+```
 
-The first run may take a few minutes while the image is being
-downloaded. After that, you'll be inside the container and you can run
-the playbooks:
+1. Set the `openstack_ssh_public_key` to your OpenStack keypair name.
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/README.md]: I see you addressed the issue I raised about the wrong key name here. Thank you.

File: playbooks/provisioning/openstack/advanced-configuration.md [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+auto evaluated in-stack DNS server's **public** IP.
+
+Note, for the in-stack DNS server, private view updates may be sent only
+via the public IP of the server. You can not send updates via the private
+IP yet. This forces the in-stack private server to have a floating IP.
+See also the [security notes](#security-notes)
+
+## Other configuration variables
+
+`openstack_ssh_key` is a Nova keypair - you can see your keypairs with
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/advanced-configuration.md]: This keyname is still wrong. I don't see this key in the sub directories. s/openstack_ssh_key/openstack_ssh_public_key/

File: playbooks/provisioning/openstack/advanced-configuration.md [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+
+
+
+## Configure static inventory and access via a bastion node
+
+Example inventory variables:
+
+    openstack_use_bastion: true
+    bastion_ingress_cidr: "{{openstack_subnet_prefix}}.0/24"
+    openstack_private_ssh_key: ~/.ssh/openshift
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/advanced-configuration.md]: This private key is not consistent with the README.md file. Both the all.yml and this file call the path ~/.ssh/openshift

File: playbooks/provisioning/openstack/README.md [hunk 4]
```diff
@@ -1,608 +1,260 @@
...
+   - See `openstack keypair list` to find the keypairs registered with
+   OpenShift.
+   - This must correspond to your private SSH key in `~/.ssh/id_rsa`
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/README.md]: This key path is inconsistent with the rest of the changes here. The majority of the files (advanced-configuration.md and all.yml comments) use "~/.ssh/openshift". Either change the other references or change this one.

File: playbooks/provisioning/openstack/README.md [hunk 5]
```diff
@@ -1,608 +1,260 @@
...
-    cp openshift-ansible-contrib/playbooks/provisioning/openstack/sample-inventory/ansible.cfg ansible.cfg
+Your *Ansible host* needs to have the following:
 
-### Update `inventory/group_vars/all.yml`
+1. Docker
+2. `keystonerc` file with your OpenStack credentials
+3. SSH private key for logging in to your OpenShift nodes
 
-#### DNS configuration variables
+Assuming your private key is `~/.ssh/id_rsa` and `keystonerc` in your
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/README.md]: Use consistent key names.

File: playbooks/provisioning/openstack/README.md [hunk 6]
```diff
@@ -1,608 +1,260 @@
...
-that ssh config, use the `-F` prefix, f.e.:
-
-    ssh -F /tmp/ssh.config.openshift.ansible.openshift.example.com master-0.openshift.example.com echo OK
-
-Note, relative paths will not work for the `openstack_ssh_config_path`, but it
-works for the `openstack_private_ssh_key` and `openstack_inventory_path`. In this
-guide, the latter points to the current directory, where you run ansible commands
-from.
+```bash
+$ less .ssh/id_rsa
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/README.md]: Consistent key names.

File: playbooks/provisioning/openstack/advanced-configuration.md [hunk 7]
```diff
@@ -0,0 +1,699 @@
...
+
+You need to download the OpenShift command line client (called `oc`).
+You can download and extract `openshift-origin-client-tools` from the
+OpenShift release page:
+
+https://github.com/openshift/origin/releases/latest/
+
+Or you can now copy it from the master node:
+
+    $ ansible --private-key ~/.ssh/openshift -i inventory masters[0] -m fetch -a "src=/bin/oc dest=oc"
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/advanced-configuration.md]: Consistent private key names.

File: playbooks/provisioning/openstack/advanced-configuration.md [hunk 8]
```diff
@@ -0,0 +1,699 @@
...
+The first infra node then becomes a bastion node as well and proxies access
+for future ansible commands. The post-provision step also configures Satellite,
+if requested, and DNS server, and ensures other OpenShift requirements to be met.
+
+## Running Custom Post-Provision Actions
+
+A custom playbook can be run like this:
+
+```
+ansible-playbook --private-key ~/.ssh/openshift -i inventory/ openshift-ansible-contrib/playbooks/provisioning/openstack/custom-actions/custom-playbook.yml
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/advanced-configuration.md]: Here you used "~/.ssh/openshift"

File: playbooks/provisioning/openstack/advanced-configuration.md [hunk 9]
```diff
@@ -0,0 +1,699 @@
...
+```
+
+If you'd like to limit the run to one particular host, you can do so as follows:
+
+```
+ansible-playbook --private-key ~/.ssh/openshift -i inventory/ openshift-ansible-contrib/playbooks/provisioning/openstack/custom-actions/custom-playbook.yml -l app-node-0.openshift.example.com
```
Comment (mbruzek) [inline playbooks/provisioning/openstack/advanced-configuration.md]: openshift++

**Comment by mbruzek** [comment]
@tomassedovic Thanks for including me on the review. I noticed that there was several inconsistent uses of the private key name across two of the markdown documents. My recommendation is to figure out which one is correct and standardize on that one. See my many comments in-line (sorry I made so many but I wanted to make sure they all were consistent).

The other content in the markdown documents looked like an improvement and would be a welcome addition to the docs.

**Comment by tomassedovic** [comment]
@mbruzek thanks! I've mostly move the old README to advanced-config and plan to cleaned that up later. But we should fix the key name and path issues. I'll take a look.

**Comment by tomassedovic** [comment]
This should address issues raised by @mbruzek as well as resolve merge conflicts.

**Comment by mbruzek** [comment]
@tomassedovic looks like you got them all in these files. Remember to address the comments in all.yml.
+1 LGTM

File: playbooks/provisioning/openstack/advanced-configuration.md [hunk 10]
```diff
@@ -0,0 +1,699 @@
...
+
+Either way, find the `oc` binary and put it in your `PATH`.
+
+
+### Logging in Using the Command Line
+
+
+```bash
+oc login --insecure-skip-tls-verify=true https://console.openshift.example.com:8443 -u user -p password
+oc new-project test
```
Comment (Tlacenka) [inline playbooks/provisioning/openstack/advanced-configuration.md]: Just a tiny visual nitpick. Maybe choose a project name that doesn't trigger highlighting in bash?

File: playbooks/provisioning/openstack/README.md [hunk 11]
```diff
@@ -1,608 +1,260 @@
...
 
-    . keystonerc
-    ansible-playbook openshift-ansible-contrib/playbooks/provisioning/openstack/provision.yaml
+Let's disable those checks by putting this in
+`inventory/group_vars/OSEv3.yml`:
 
-Note, here you start with an empty inventory. The static inventory will be populated
-with data so you can omit providing additional arguments for future ansible commands.
+```yaml
+openshift_disable_check: disk_availability,memory_availability
```
Comment (Tlacenka) [inline playbooks/provisioning/openstack/README.md]: Not sure if there's a list of all options somewhere, but a link to it could be helpful? Or adding some of them to the advanced configuration file? e. g. While I was trying it out, I had to add `docker_storage` there.

File: playbooks/provisioning/openstack/sample-inventory/group_vars/OSEv3.yml [hunk 12]
```diff
@@ -5,8 +5,8 @@ openshift_deployment_type: origin
 openshift_master_default_subdomain: "apps.{{ env_id }}.{{ public_dns_domain }}"
 
 openshift_master_cluster_method: native
-openshift_master_cluster_hostname: "{{ groups.lb.0|default(groups.masters.0) }}"
-openshift_master_cluster_public_hostname: "{{ groups.lb.0|default(groups.masters.0) }}"
+openshift_master_cluster_public_hostname: "console.{{ env_id }}.{{ public_dns_domain }}"
+openshift_master_cluster_hostname: "{{ openshift_master_cluster_public_hostname }}"
```
Comment (bogdando) [inline playbooks/provisioning/openstack/sample-inventory/group_vars/OSEv3.yml]: could you move these into a separate PR? otherwise, LGTM

File: ci/openstack/validate.sh [hunk 13]
```diff
@@ -23,13 +23,13 @@ function restore_dns {
...
 mkdir -p bin
-scp -o "StrictHostKeyChecking no" -o "UserKnownHostsFile /dev/null" openshift@master-0.$ENV_ID.example.com:/usr/bin/oc bin/
+scp -o "StrictHostKeyChecking no" -o "UserKnownHostsFile /dev/null" openshift@console.$ENV_ID.example.com:/usr/bin/oc bin/
 ls -alh bin
 
 export PATH="$PWD/bin:$PATH"
 ENV_ID="openshift-$TRAVIS_BUILD_NUMBER"
 
-oc login --insecure-skip-tls-verify=true https://master-0.$ENV_ID.example.com:8443 -u test -p password
+oc login --insecure-skip-tls-verify=true https://console.$ENV_ID.example.com:8443 -u test -p password
```
Comment (bogdando) [inline ci/openstack/validate.sh]: looks like an unrelated change
